src/x11.cc

```diff
@@ -1205,61 +1084,83 @@ void print_desktop_name(struct text_object *obj, char *p,
 }
 
 #ifdef OWN_WINDOW
+enum class x11_strut : size_t {
+  LEFT,
+  RIGHT,
+  TOP,
+  BOTTOM,
+  LEFT_START_Y,
+  LEFT_END_Y,
+  RIGHT_START_Y,
+  RIGHT_END_Y,
+  TOP_START_X,
+  TOP_END_X,
+  BOTTOM_START_X,
+  BOTTOM_END_X,
+  COUNT
+};
+constexpr size_t operator*(x11_strut index) {
+  return static_cast<size_t>(index);
+}
+
 /* reserve window manager space */
-void set_struts(int sidenum) {
-  Atom strut;
-  if ((strut = ATOM(_NET_WM_STRUT)) != None) {
+void set_struts(alignment align) {
+  // Middle and none align don't have least significant bit set.
+  // Ensures either vertical or horizontal axis are start/end
+  if ((*align & 0b0101) == 0) return;
```

Comment:
I added a check to verify `alignement` is actually valid, i.e. would reserve space on one of 4 screen sides.

This is a cheap check (no jumps), and works because both left/top and right/right have least significant bit set to `1`, while none and middle don't. `alignment` is `vertical << 2 + horizontal`, so we can just check `0x0101` pattern to see whether _either_ is set.

Reply:
This check avoids calls to X11 and jumps if the `alignment` doesn't do anything.